### ISSUE TITLE
Make it possible for admins to add third-party members to third-party groups

### ISSUE THREAD

**Issue by lyzadanger** [hypothesis/h#6109]
We have a customer who is using restricted groups on a third-party authority. That means the `creator` of the group and all of its members need to be within that third-party authority.

The creator and members fields accept usernames (not fully qualified account ids). Right now the creator field validation https://github.com/hypothesis/h/blob/92e42a34669c9d051ad2e6a9339c6df0bff41576/h/schemas/forms/admin/group.py#L28 validates against the group's authority—OK, that works. But the validation for member usernames acts against the default authority https://github.com/hypothesis/h/blob/92e42a34669c9d051ad2e6a9339c6df0bff41576/h/schemas/forms/admin/group.py#L72 — that is a problem!

Make it so members can be added to both first- and third-party groups from the form at `https://hypothes.is/admin/groups/:groupid`

**Comment by jon-betts** [comment]
Just had a call with @lyzadanger to try and get my head straight on this. To summarise:

**Background**

 * Groups exist in organisations (many:1)
 * Organisations exist in authorities (many:1)
 * The default authority is us (1st party) anything else is 3rd party
 * When adding a creator to a group, in an org, we do not fully qualify the username in the form
 * The authority is implied by the choice of organisation
 * Therefore we check in the validation that the `username@<implied authority>` exists
 * All of this behavior is correct

**The problem**

* You can also use this form to add members to the group
* These members also are not fully qualified
* **But** when they are validated, they are always validated against the default Hypothes.is authority
* This works fine if the group is a 1st party group, as everything lines up
* This is a bug for 3rd party groups and the subject of this ticket

**Comment by jon-betts** [comment]
I don't think this can be done as suggested. The validators appear to be attached to various nodes in the hierarchy of the data. At which point they have access to the data from that node downwards.

As checking the creator requires you to know the organisation, the validator for that item is attached at the root so it can see both. The validator for members is attached at the individual member level, and so can't see the value chosen for the group.

I think what I'll do is move the validation of the members inside the root validator if possible.
